### ISSUE TITLE
wrong timing recordings

### ISSUE THREAD

**Issue by JeanMertz** [ArturT/knapsack#9]
Whenever we run our cukes (roughly 3 minutes to run), we get this (anonymized) output in Travis:

```
554 scenarios (554 passed)
4545 steps (4545 passed)
2m6.315s


Knapsack report was generated. Preview:
{
  "features/**/*.feature": 1.003159761428833,
  "features/**/*.feature": 1.196033239364624,
  "features/**/recommend.feature": 57018183.49009848,
  "features/**/*.feature": 0.2130286693572998,
  "features/**/*.feature": 0.2160334587097168,
  "features/**/*.feature": 0.45173168182373047,
  "features/**/*.feature": 0.04832601547241211,
  "features/**/*.feature": 1.230623483657837,
  "features/**/*.feature": 0.7682235240936279,
  "features/**/*.feature": 0.10379314422607422,
  "features/**/*.feature": 0.5373373031616211,
  "features/**/*.feature": 2.7034032344818115,
  "features/**/*.feature": 0.32158589363098145,
  "features/**/*.feature": 0.19092035293579102,
  "features/**/*.feature": 0.6193909645080566,
  "features/**/*.feature": 0.5318620204925537,
  "features/**/*.feature": 0.7692575454711914,
  "features/**/*.feature": 0.7230222225189209,
  "features/**/*.feature": 0.495070219039917,
  "features/**/*.feature": 0.4870891571044922,
  "features/**/*.feature": 0.20952224731445312,
  "features/**/*.feature": 0.7473139762878418,
  "features/**/*.feature": 0.47346997261047363,
  "features/**/*.feature": 1.0329043865203857,
  "features/**/*.feature": 1.8328258991241455,
  "features/**/*.feature": 0.8722712993621826,
  "features/**/*.feature": 1.2147345542907715,
  "features/**/*.feature": 0.48812413215637207,
  "features/**/*.feature": 0.42723989486694336,
  "features/**/*.feature": 0.9765002727508545,
  "features/**/*.feature": 0.8365063667297363,
  "features/**/*.feature": 1.4077143669128418,
  "features/**/*.feature": 0.029282331466674805,
  "features/**/*.feature": 0.11276030540466309,
  "features/**/*.feature": 0.5681333541870117,
  "features/**/*.feature": 0.38565659523010254,
  "features/**/*.feature": 1.343261480331421,
  "features/**/*.feature": 0.6901326179504395,
  "features/**/*.feature": 1.1266040802001953,
  "features/**/*.feature": 2.198932647705078,
  "features/**/*.feature": 0.11878585815429688,
  "features/**/*.feature": 0.27373504638671875,
  "features/**/*.feature": 3.770233631134033,
  "features/**/*.feature": 0.2858238220214844,
  "features/**/*.feature": 0.5811190605163574,
  "features/**/*.feature": 0.6995368003845215,
  "features/**/*.feature": 0.6289098262786865,
  "features/**/*.feature": 0.4094107151031494,
  "features/**/*.feature": 0.74979567527771,
  "features/**/email.feature": 5391999.136254549,
  "features/**/*.feature": 0.33169007301330566,
  "features/**/*.feature": 0.2166881561279297,
  "features/**/*.feature": 0.03311800956726074,
  "features/**/*.feature": 0.030640840530395508,
  "features/**/*.feature": 0.17110943794250488,
  "features/**/*.feature": 0.48764514923095703,
  "features/**/*.feature": 0.8722720146179199,
  "features/**/*.feature": 0.555567741394043,
  "features/**/*.feature": 0.9950838088989258,
  "features/**/*.feature": 0.25239014625549316,
  "features/**/*.feature": 0.12649965286254883,
  "features/**/*.feature": 0.1031959056854248,
  "features/**/*.feature": 0.029646635055541992,
  "features/**/*.feature": 0.1288614273071289,
  "features/**/*.feature": 4.771716833114624,
  "features/**/export.feature": 26731273.147525072,
  "features/**/*.feature": 0.10891866683959961,
  "features/**/*.feature": 0.6129484176635742,
  "features/**/*.feature": 0.5339069366455078,
  "features/**/*.feature": 2.2400553226470947,
  "features/**/*.feature": 3.9317946434020996,
  "features/**/*.feature": 1.2024469375610352,
  "features/**/*.feature": 1.5093786716461182,
  "features/**/*.feature": 0.39945054054260254,
  "features/**/*.feature": 0.529149055480957,
  "features/**/*.feature": 0.4336683750152588,
  "features/**/*.feature": 1.2973487377166748,
  "features/**/*.feature": 0.11217498779296875,
  "features/**/*.feature": 1.6423816680908203,
  "features/**/*.feature": 0.3878803253173828,
  "features/**/*.feature": 0.5852077007293701,
  "features/**/*.feature": 1.1968224048614502,
  "features/**/*.feature": 1.0070815086364746,
  "features/**/*.feature": 1.324880599975586,
  "features/**/*.feature": 0.7847650051116943,
  "features/**/*.feature": 0.15419507026672363,
  "features/**/*.feature": 0.1375415325164795,
  "features/**/*.feature": 0.2437891960144043,
  "features/**/*.feature": 0.038895606994628906,
  "features/**/*.feature": 0.07857513427734375,
  "features/**/*.feature": 1.3422677516937256,
  "features/**/*.feature": 0.472412109375,
  "features/**/*.feature": 2.7329916954040527,
  "features/**/*.feature": 0.029615402221679688,
  "features/**/*.feature": 0.7670252323150635,
  "features/**/*.feature": 0.21507048606872559,
  "features/**/search.feature": 7807642.681318045,
  "features/**/*.feature": 3.499208450317383,
  "features/**/*.feature": 0.235215425491333,
  "features/**/*.feature": 0.5279958248138428,
  "features/**/*.feature": 0.21576786041259766,
  "features/**/*.feature": 0.09094381332397461,
  "features/**/*.feature": 0.37752389907836914,
  "features/**/*.feature": 1.363354206085205,
  "features/**/*.feature": 3.0626842975616455,
  "features/**/*.feature": 0.1384115219116211,
  "features/**/*.feature": 7.06137752532959,
  "features/**/*.feature": 1.7372627258300781,
  "features/**/*.feature": 1.0493583679199219,
  "features/**/*.feature": 0.25408387184143066,
  "features/**/*.feature": 4.52888560295105,
  "features/**/*.feature": 0.5256309509277344,
  "features/**/*.feature": 0.2931356430053711,
  "features/**/*.feature": 1.2082080841064453,
  "features/**/*.feature": 0.5562317371368408,
  "features/**/*.feature": 1.2475011348724365,
  "features/**/*.feature": 0.3939666748046875,
  "features/**/*.feature": 0.13701748847961426,
  "features/**/*.feature": 0.28044581413269043,
  "features/**/*.feature": 0.10454559326171875,
  "features/**/*.feature": 7.583114147186279,
  "features/**/*.feature": 3.2411201000213623,
  "features/**/*.feature": 0.423907995223999,
  "features/**/widget.feature": 7807685.043151617
}
Knapsack global time execution for tests: 11h 08m 15s
```

While some of those features take a while longer, they don't take *that* much longer, and also the "global time execution" reported by Knapsack is a long way off compared to the real Travis run.

**Comment by ArturT** [comment]
This output is generated only when you set KNAPSACK_GENERATE_REPORT=true.

The KNAPSACK_GENERATE_REPORT variable should be set only when you want to generate report for the first time (or when you want to regenerate report, see https://github.com/ArturT/knapsack#adding-or-removing-tests ) which should be stored in file `knapsack_cucumber_report.json` and commited into repository. 
You want to generate report just once to collect info about all tests. Please run it on single travis machine.
After you commited report into your repository you can update .travis.yml to use knapsack:cucumber rake task, see https://github.com/ArturT/knapsack#info-for-travis-users

When you want to run command `bundle exec rake knapsack:cucumber` you should not set KNAPSACK_GENERATE_REPORT.

**Comment by JeanMertz** [comment]
@ArturT I am aware of all the points you raised. Maybe I didn't express my problem clear enough.

I did exactly as described in the README. I first ran the test suite on Travis *with* `KNAPSACK_GENERATE_REPORT=true`, this resulted in the above output.

Now, because of the skewed output (extreme high numbers for some feature files and a weird "global time execution" number (11+ hours?!)), the tests aren't distributed evenly, still causing a slower than needed test suite.

**Comment by ArturT** [comment]
Oh, I see. It looks like a few test files have too long time execution:
```
"features/**/recommend.feature": 57018183.49009848,
"features/**/email.feature": 5391999.136254549,
"features/**/export.feature": 26731273.147525072,
"features/**/export.feature": 26731273.147525072,
"features/**/search.feature": 7807642.681318045,
"features/**/widget.feature": 7807685.043151617
```
I'm curious how tests look like in those files. For now cucumber adapter covers regular scenario an scenario outlines https://github.com/ArturT/knapsack/blob/master/lib/knapsack/adapters/cucumber_adapter.rb#L7
Probably I missed something and tests are measured in a wrong way. It would be nice if you could share code of the test file with me or if you could try to debug `bind_time_tracker` and `CucumberAdapter.test_path` method.

**Comment by ArturT** [comment]
@JeanMertz Let me know how is going. It would be nice if we could somehow figure out what's the reason of such big time execution.

**Comment by ArturT** [comment]
@JeanMertz Hi, did you figure out how to solve the problem? Can I somehow help?

**Comment by ArturT** [comment]
@JeanMertz There is probably another reason of this bug, maybe those specs use Timecop gem? Could you check that? Thanks. https://github.com/ArturT/knapsack/issues/11

**Comment by ArturT** [comment]
I fixed the issue #11 which probably is the reason of the bug you experienced. Please check new gem version. I'm closing this issue.
